snapcraft.yaml

```diff
@@ -5,54 +5,74 @@ description: >
   HexChat is an easy to use yet extensible IRC Client. It allows you to securely join multiple networks and talk to users privately or in channels using a customizable interface. You can even transfer files.
 confinement: strict
 grade: stable
-# FIXME: Platforms are not auto-downloaded
-#plugs:
-#  gnome-3-26-1604:
-#    interface: content
-#    content: gnome-3-26-1604
-#    target: gnome-platform
-#    default-provider: gnome-3-26-1604
+base: core18
+plugs:
+  gtk-2-engines:
+    interface: content
+    target: $SNAP/lib/gtk-2.0
+    default-provider: gtk2-common-themes
+  gtk-2-themes:
+    interface: content
+    target: $SNAP/share/themes
+    default-provider: gtk2-common-themes
+  icon-themes:
+    interface: content
+    target: $SNAP/share/icons
+    default-provider: gtk-common-themes
+
+environment:
+  GTK_PATH: $SNAP/lib/gtk-2.0
+  GTK_DATA_PREFIX: $SNAP
+  XDG_DATA_DIRS: $SNAP/share:$XDG_DATA_DIRS
+
 slots:
   hexchat:
     interface: dbus
     bus: session
     name: org.hexchat.service
 apps:
   hexchat:
-    command: desktop-launch hexchat-wrapper
+    command: desktop-launch hexchat
     desktop: share/applications/io.github.Hexchat.desktop
     plugs:
-      - unity7
+      - desktop
       - desktop-legacy
-      - 'x11'
 #      - pulseaudio
       - network
       - network-bind
-#      - gnome-3-26-1604
+      - x11
 parts:
-#  desktop-gnome-platform:
-#    stage:
-#      # We override this in hexchat-copy with gtk2 version
-#      - -flavor-select
-  meson:
-    source: https://github.com/mesonbuild/meson/releases/download/0.45.0/meson-0.45.0.tar.gz
-    plugin: python
-    prime: [-*]
+  desktop-gtk2:
+    source: https://github.com/ubuntu/snapcraft-desktop-helpers.git
+    source-subdir: gtk
+    plugin: make
+    make-parameters: ["FLAVOR=gtk2"]
+    build-packages:
+      - libgtk2.0-dev
+    stage-packages:
+      - libxkbcommon0  # XKB_CONFIG_ROOT
+      - shared-mime-info
+      - libgtk2.0-0
+      - libgdk-pixbuf2.0-0
+      - libglib2.0-bin
+      - libgtk2.0-bin
+      - locales-all
+      - xdg-user-dirs
+        
   hexchat:
 # FIXME: Spell check needs dictionaries
 # FIXME: Notifications can't see the desktop file or icon
 # FIXME: Platform see above:
```

Comment:
So this still isn't using a "platform" right? Most things are still bundled?

Reply:
it uses the themeing (engines, themes and icons) platform snap, not the gtk libs temselves though